servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/ReadOnlyTcpServerConfig.java

```diff
@@ -16,75 +16,22 @@
 
 import java.net.SocketOption;
 import java.time.Duration;
 import java.util.Map;
-import java.util.Map.Entry;
 import javax.annotation.Nullable;
 
-import static io.servicetalk.transport.api.TransportObservers.asSafeObserver;
-import static io.servicetalk.transport.netty.internal.SslContextFactory.forServer;
-
 /**
  * Read only view of {@link TcpServerConfig}.
  */
-public final class ReadOnlyTcpServerConfig extends AbstractReadOnlyTcpConfig<ServerSslConfig> {
-    @SuppressWarnings("rawtypes")
-    private final Map<ChannelOption, Object> listenOptions;
-    private final TransportObserver transportObserver;
-    @Nullable
-    private final ServerSslConfig sslConfig;
-    @Nullable
-    private final SslContext sslContext;
-    @Nullable
-    private final Mapping<String, SslContext> sniMapping;
-    private final int sniMaxClientHelloLength;
-    private final Duration sniClientHelloTimeout;
-    private final boolean alpnConfigured;
-
-    ReadOnlyTcpServerConfig(final TcpServerConfig from) {
-        super(from);
-        listenOptions = nonNullOptions(from.listenOptions());
-        final TransportObserver transportObserver = from.transportObserver();
-        this.transportObserver = transportObserver == NoopTransportObserver.INSTANCE ? transportObserver :
-                asSafeObserver(transportObserver);
-        this.sslConfig = from.sslConfig();
-        final Map<String, ServerSslConfig> sniMap = from.sniConfig();
-        if (sniMap != null) {
-            if (sslConfig == null) {
-                throw new IllegalStateException("No default security config defined but found SNI config mappings");
-            }
-            sslContext = forServer(sslConfig);
-            boolean foundAlpn = !sslContext.applicationProtocolNegotiator().protocols().isEmpty();
-            final DomainWildcardMappingBuilder<SslContext> mappingBuilder =
-                    new DomainWildcardMappingBuilder<>(sniMap.size(), sslContext);
-            for (Entry<String, ServerSslConfig> sniConfigEntry : sniMap.entrySet()) {
-                SslContext sniContext = forServer(sniConfigEntry.getValue());
-                foundAlpn |= !sniContext.applicationProtocolNegotiator().protocols().isEmpty();
-                mappingBuilder.add(sniConfigEntry.getKey(), sniContext);
-            }
-            sniMapping = mappingBuilder.build();
-            alpnConfigured = foundAlpn;
-        } else if (sslConfig != null) {
-            sslContext = forServer(sslConfig);
-            sniMapping = null;
-            alpnConfigured = !sslContext.applicationProtocolNegotiator().protocols().isEmpty();
-        } else {
-            sslContext = null;
-            sniMapping = null;
-            alpnConfigured = false;
-        }
-        sniMaxClientHelloLength = from.sniMaxClientHelloLength();
-        sniClientHelloTimeout = from.sniClientHelloTimeout();
-    }
+public interface ReadOnlyTcpServerConfig extends ReadOnlyTcpConfig<ServerSslConfig> {
```

Comment:
Even though this one is in `-internal` module, it was still public class. We can not make such changes in 0.42.x branch.